library/alloc/src/boxed.rs

```diff
@@ -1330,6 +1331,12 @@ impl<T: Copy> From<&[T]> for Box<[T]> {
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "box_from_cow", since = "1.45.0")]
 impl<T: Copy> From<Cow<'_, [T]>> for Box<[T]> {
+    /// Converts a `Cow<'_, [T]>` into a `Box<[T]>`
+    ///
+    /// When `cow` is the `Cow::Borrowed` variant, this
+    /// conversion allocates on the heap and performs a the
+    /// underlying `str`. Otherwise, it re-used the owned
+    /// string.
```

Comment:
There's a missing "copy" and a typo in "used". "string" is also wrong here (probably copied from the `Box<str>` docs?).

Also, technically it's not guaranteed to reuse the `Vec`'s allocation since it will end up calling `Vec::into_boxed_slice`, which will reallocate if the vec has spare capacity. It might, but it's not guaranteed.

```suggestion
    /// When `cow` is the `Cow::Borrowed` variant, this
    /// conversion allocates on the heap and performs a copy the
    /// underlying `str`. Otherwise, it will try to re-use the owned
    /// vec's allocation.
```

Reply:
Thank you for your close read. Should we capitalize vec in "owned vec's allocation"?

Reply:
I should also change that to "a copy _of_ the underlying.."

Reply:
```suggestion
    /// When `cow` is the `Cow::Borrowed` variant, this
    /// conversion allocates on the heap and performs a copy of the
    /// underlying `str`. Otherwise, it will try to re-use the owned
    /// vec's allocation.
```